### ISSUE TITLE
[Bug]: JavaScript parsing error: 'import', and 'export' cannot be used outside of module code

### ISSUE THREAD

**Issue by yinhunfeixue** [web-infra-dev/rspack#7887]
### System Info

system: macos 15.0
node: 20.16.0

### Details

[rspack.zip](https://github.com/user-attachments/files/16989814/rspack.zip)

**操作步骤：**
1. 解压附件文件
2. `npm i`
3. `node pack.js`   

`JavaScript parsing error: 'import', and 'export' cannot be used outside of module code`

**问题：**
`mode: "development"` 正常
`mode:  "production",` 报错
<img width="1083" alt="image" src="https://github.com/user-attachments/assets/6ca24536-8ba8-4427-ac24-3b8a530c7a04">

### Reproduce Steps

see details

**Comment by inottn** [comment]
Rspack uses the SwcJsMinimizerRspackPlugin by default to compress code, and you may need to configure the minimizerOptions to set `module` to true.

**Comment by hardfist** [comment]
> Rspack uses the SwcJsMinimizerRspackPlugin by default to compress code, and you may need to configure the minimizerOptions to set `module` to true.

this is not the best solution since it will cause all js assets are treated module, you can control specific assets by set assetInfo
```js
 compilation.updateAsset(entry, new ConcatSource(
          `export default function(global, globalThis, window, $app_exports$, $app_evaluate$){`,
          source,
          `}`
        ), {
          javascriptModule:true // set this asset as module
  });
```

**Comment by hardfist** [comment]
should be fixed in #7924

**Comment by hardfist** [comment]
> > Rspack uses the SwcJsMinimizerRspackPlugin by default to compress code, and you may need to configure the minimizerOptions to set `module` to true.
> 
> this is not the best solution since it will cause all js assets are treated module, you can control specific assets by set assetInfo
> 
> ```js
>  compilation.updateAsset(entry, new ConcatSource(
>           `export default function(global, globalThis, window, $app_exports$, $app_evaluate$){`,
>           source,
>           `}`
>         ), {
>           javascriptModule:true // set this asset as module
>   });
> ```
> 
> but rspack has some bugs about dealing with assetInfo, I'm gonna fix it

should be fixed in should be fixed in #7924
